### ISSUE TITLE
bad corrupt seen with broadway

### ISSUE THREAD

**Issue by shi-yan** [mbebenita/Broadway#71]
I'm working on my video streaming project. 

I noticed very bad corruption with broadway decoder. such a problem is not seen with VLC player and the same video file. I'm wondering if there is a known issue?

**Comment by shi-yan** [comment]
I'm not sure what the real problem is, but I noticed that if I let the number of reference image (encoder setting) be 1, the corruption will be gone. before, it was 2.

**Comment by soliton4** [comment]
i am pretty sure its because you have to feed complete nal units to the decode function

http://en.wikipedia.org/wiki/Network_Abstraction_Layer#VCL_and_Non-VCL_NAL_Units

feel free to copy paste this implementation until i provide a nice version for broadway
https://github.com/soliton4/nodeMirror/blob/edd903a9a75728833da714890e136776796994ff/src/avc/Wgt.js#L291

**Comment by shi-yan** [comment]
thank you very much for replaying. but what you have guessed was wrong. 

I have used http://h264bitstream.sourceforge.net/ to separate my nal packets. I'm very sure I'm feeding the entire nals. more over, I printed the nal sizes after encoding, and compared them once they are received by javascript. 

This issue is not broadway only. I can reproduce this problem with vlc and certain decoders. For example, with vaapi decoder on linux and hw decoder on mac. But it won't repro with VDPAU.
